hydroflow/src/builder/hydroflow_builder.rs

```diff
@@ -109,6 +112,8 @@ impl HydroflowBuilder {
                 pivot.run();
             },
         );
+        assert_eq!(sg_id, sg_id_actual);
```

Comment:
Changes here

Reply:
I can't think of a better way to do this with the current architecture but it seems unfortunate that we have to do this concurrently and then compare them :(

Reply:
I agree